File: deleteMiddleNodeofLL.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        newNode= ListNode(next=head)
        slow= newNode
        fast=head
        while(fast and fast.next):
            slow= slow.next
            fast= fast.next.next
        slow.next= slow.next.next
        return newNode.next

File: DeleteMiddleNodeOfLL.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* newNode = new ListNode(0, head);
        ListNode* s = newNode;
        ListNode* f = head;
        while (f && f->next) {
            s = s->next;
            f = f->next->next;
        }
        s->next = s->next->next;
        return newNode->next;
    }
};
        
  
